src/instruction/jk_inst.rs

```diff
@@ -3,30 +3,49 @@
 //! really an `Instruction`, and therefore their implementation lives in the parser
 //! module. They are executed at "compile" time, when running through the code first.
 
-use crate::instruction::{InstrKind, Instruction};
+use crate::instruction::{InstrKind, Instruction, FunctionCall};
 use crate::{Interpreter, JkErrKind, JkError};
 
 /// The potential interpreter instructions
 #[derive(Clone, Debug, PartialEq)]
-pub enum JkInst {
+pub enum JkInstKind {
     Dump,
     Quit,
+    Ir,
+}
+
+#[derive(Clone)]
+pub struct JkInst {
+    kind: JkInstKind,
+    args: Vec<Box<dyn Instruction>>,
 }
 
 impl JkInst {
-    /// Construct a `JkInst` from a given keyword
-    pub fn from_str(keyword: &str) -> Result<Self, JkError> {
-        match keyword {
-            "dump" => Ok(JkInst::Dump),
-            "quit" => Ok(JkInst::Quit),
+    /// Construct a `JkInst` from a `FunctionCall`
+    pub fn from_function_call(fc: FunctionCall) -> Result<Self, JkError> {
+        let func_name = fc.name();
+
+        let kind = match func_name {
+            "dump" => JkInstKind::Dump,
+            "quit" => JkInstKind::Quit,
+            "ir" => JkInstKind::Ir,
             // FIXME: Fix location
-            _ => Err(JkError::new(
+            _ => return Err(JkError::new(
                 JkErrKind::Parsing,
-                format!("unknown interpreter directive @{}", keyword),
+                format!("unknown interpreter directive @{}", func_name),
                 None,
-                keyword.to_owned(),
+                func_name.to_owned(),
             )),
-        }
+        };
+
+        Ok(Self {
+            kind,
+            args: fc.args().to_owned()
```

Comment:
```suggestion
            args: fc.args().clone()
```
I think the behavior is the same but I might be mistaken. If it is the same, then I think this version is clearer and shows the intent better